### DISCUSSION TITLE
Is there any way to prevent swiping when user intereacts with specific dom object?

### DISCUSSION THREAD

**Discussion by async3619** [naver/egjs-flicking#696] [Q&A] [answered]
In swiperjs, there's [noSwipingClass](https://swiperjs.com/swiper-api#param-noSwipingClass) and [noSwipingSelector](https://swiperjs.com/swiper-api#param-noSwipingSelector) configuration property that can prevent swiping when user intereacts with specified target.

Is there any similar thing in egjs-flicking?

**Comment by async3619** [comment]
oh yeah I found `onHoldStart` event and It can prevent swiping. but since this event is called before other events, it seems difficult to block swiping when accessing an object inside the slide.

**Comment by malangfox** [comment, marked as answer]
Hello @async3619.

Among Flicking's options, there is no separate option that supports this feature.
It seems that blocking the mousedown and touchdown event propagation in the child element inside Flicking can temporarily solve this issue.
We will discuss whether we can support this feature in the future.

Thank you.

**Reply by async3619** [reply]
방법을 찾은 것 같습니다. moveStart 이벤트 이용해서 swipe 단계에서 intercept 하면 대응이 가능한것 같습니다.

```ts
private handleMoveStart = (e: MoveStartEvent<Flicking>) => {
    if (
        !e.axesEvent.inputEvent ||
        !("srcEvent" in e.axesEvent.inputEvent) ||
        !(e.axesEvent.inputEvent.srcEvent instanceof TouchEvent)
    ) {
        return;
    }

    const path = e.axesEvent.inputEvent.srcEvent.composedPath();
    for (const target of path) {
        if (!(target instanceof HTMLElement)) {
            continue;
        }

        if (target.classList.contains("no-swipe")) {
            e.stop();
        }
    }
};
```

다만 type definition이 부재해서 사용하면서도 이 방법이 과연 맞는건가 의심이 들더라구요. 도움 주셔서 감사합니다.
